module/modmanager/manager.go

```diff
@@ -488,6 +488,18 @@ func (mgr *Manager) addResource(ctx context.Context, conf resource.Config, deps
 
 	_, err = mod.client.AddResource(ctx, &pb.AddResourceRequest{Config: confProto, Dependencies: deps})
 	if err != nil {
+		// if there is an error while adding a modular resource, send a RemoveResource call to attempt clean up.
```

Comment:
still convincing myself that this would not be a regression, we could technically match on the specific error (`resource already exists`) but that feels more brittle. Doing this only on context done will not fix issues since the resource would not have been created yet

Reply:
removed the change since it feels too heavy handed - happy to discuss

Reply:
What issue could we run into calling `RemoveResource` if `AddResource` fails? An error from the module saying that the resource doesn't exist? Should we just `UncheckedError` it? I'm not sure I see what what you had was too heavy-handed.

Reply:
the main concern would probably be removing a resource that does exist - should the correct behavior in those cases be to remove then add, or just reconfigure?

Also might be a moot point if all the sdks handle cancels/timeouts properly

Reply:
> should the correct behavior in those cases be to remove then add, or just reconfigure?

I think the only way we can say that a modular resource has "properly built" is if `AddResource` from the RDK returns no error. If `AddResource` failed for the RDK but succeeded for the module, I don't think the RDK would be in a position to handle reconfiguring the modular resource. In my mental model, it would make the most sense to remove, ignore and log any returned errors, then try adding again at a later point.

> Also might be a moot point if all the sdks handle cancels/timeouts properly

Agree. If SDK module libraries handle cancels/timeouts properly, and `close` their respective resources, I think `Remove` is redundant. I'd be fine if you don't want to use `RemoveResource` here and rely on https://viam.atlassian.net/browse/RSDK-8424 being implemented correctly and somewhat soon.

Reply:
ya, let's wait for more bug reports before fixing. I agree that in theory there shouldn't be much of an issue to add the additional Remove